### PR TITLE
pulumi: init at 0.16.2

### PR DESCRIPTION
**Pull request by peterromfeldhk**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by peterromfeldhk** [comment]
i tried to build from source but it does so much crap with `pip` + `yarn` and maybe more that i gave up and fallback to just fetch the binary

File: pkgs/tools/admin/pulumi/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    else throw "platform not ${stdenv.system} supported!";
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp * $out/bin/
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = https://pulumi.io/;
+    description = "Pulumi is a cloud development platform that makes creating cloud programs easy and productive.";
```
Comment (c0bw3b) [inline pkgs/tools/admin/pulumi/default.nix]: No period at the end of the short description
https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

File: pkgs/tools/admin/pulumi/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchurl, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "pulumi-${version}";
+  version = "v0.16.2";
```
Comment (c0bw3b) [inline pkgs/tools/admin/pulumi/default.nix]:
```suggestion
  version = "0.16.2";
```

>  The version part of the name attribute _must_ start with a digit (following a dash)

From https://nixos.org/nixpkgs/manual/#sec-package-naming

Then use `v${version}` in the URLs

**Review by c0bw3b — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/pulumi/default.nix (hunk 1 above)
Comment (peterromfeldhk) [reply pkgs/tools/admin/pulumi/default.nix]: there is

File: pkgs/tools/admin/pulumi/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchurl, makeWrapper }:
```
Comment (veprbl) [inline pkgs/tools/admin/pulumi/default.nix]: If `makeWrapper` is not used, please drop it

File: pkgs/tools/admin/pulumi/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+stdenv.mkDerivation rec {
+  name = "pulumi-${version}";
+  version = "0.16.2";
+
+  src = if (stdenv.system == "x86_64-linux")
```
Comment (veprbl) [inline pkgs/tools/admin/pulumi/default.nix]: `stdenv.hostPlatform.system` should work better in cross-compilation. Also multiple `if` statements can be nicely refactored as lookup in attrset (can look how it's done in `firefox-bin` or `oraclejdk`).

File: pkgs/tools/admin/pulumi/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    then fetchurl {
+      url = "https://get.pulumi.com/releases/sdk/pulumi-v${version}-linux-x64.tar.gz";
+      sha256 = "16qgy2pj3xkf1adi3882fpsl99jwsm19111fi5vzh1xqf39sg549";
+    }
+    else if (stdenv.system == "x86_64-darwin") then fetchurl {
+      url = "https://get.pulumi.com/releases/sdk/pulumi-v${version}-darwin-x64.tar.gz";
+      sha256 = "18ck9khspa0x798bdlwk8dzylbsq7s35xmla8yasd9qqlab1yy1a";
+    }
+    else throw "platform not ${stdenv.system} supported!";
```
Comment (veprbl) [inline pkgs/tools/admin/pulumi/default.nix]: Perhaps `"platform ${stdenv.system} is not supported!"`

File: pkgs/tools/admin/pulumi/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp * $out/bin/
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (veprbl) [inline pkgs/tools/admin/pulumi/default.nix]: Missing `meta.platforms` attribute

File: pkgs/tools/admin/pulumi/default.nix (hunk 4 above)
Comment (peterromfeldhk) [reply pkgs/tools/admin/pulumi/default.nix]: thanks! great feedback!

File: pkgs/tools/admin/pulumi/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  name = "pulumi-${version}";
+  version = "0.16.2";
+
+  pulumiArchPackage = {
```
Comment (Ekleog) [inline pkgs/tools/admin/pulumi/default.nix]: Putting this in the `mkDerivation` will mean potential useless rebuilds, eg. rebuilding `x86_64-linux` even if only `x86_64-darwin` changes.

File: pkgs/tools/admin/pulumi/default.nix [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    "x86_64-linux" = {
+      url = "https://get.pulumi.com/releases/sdk/pulumi-v${version}-linux-x64.tar.gz";
```
Comment (Ekleog) [inline pkgs/tools/admin/pulumi/default.nix]: Can you add a comment as to where you found these links, for easier upgrade when a new version comes out? I'm a bit surprised it doesn't support `aarch64-linux`, but why not, I guess.

Comment (peterromfeldhk) [reply pkgs/tools/admin/pulumi/default.nix]: crazy spa.. you have to switch the dropdown to manual: https://pulumi.io/quickstart/install.html

Comment (peterromfeldhk) [reply pkgs/tools/admin/pulumi/default.nix]: i initially tried to build form source.. but it got too painful after having to deal with `yarn` + `pip` in addition to `go`

Comment (Ekleog) [reply pkgs/tools/admin/pulumi/default.nix]: Can you add the `switch the dropdown to “manual” on https://pulumi.io/quickstart/install.html` comment to the source? Especially as it's unintuitive :)

As for the fact it's not built from source, then `pulumi` should be replaced with `pulumi-bin` in `all-packages.nix`, as the convention appears to be on nixpkgs for packages that are not built from source while they could.

Comment (peterromfeldhk) [reply pkgs/tools/admin/pulumi/default.nix]: i see, ill soon get back to it in more detail when sanitizing our k8s config (hope this one will do it)... wanna add update script and also need providers, was thinking to do it similar to terraform once i got time

**Review by c0bw3b — APPROVED** [review]

**Comment by Ekleog** [comment]
@GrahamcOfBorg build pulumi-bin

**Comment by Ekleog** [comment]
@peterromfeldhk Thank you! :)

**Pull request merged**